src/NServiceBus.Persistence.DynamoDB/Config/DefaultDynamoDbClientProvider.cs

```diff
@@ -0,0 +1,37 @@
+namespace NServiceBus.Persistence.DynamoDB
+{
+    using System;
+    using System.Threading;
+    using Amazon.DynamoDBv2;
+    using Amazon.Runtime;
+
+    sealed class DefaultDynamoDbClientProvider : IProvideDynamoDBClient
+    {
+        public IAmazonDynamoDB Client => client.Value;
```

Comment:
What is the best way to dispose this client? Probably not a big deal if we don't but the type implements `IDisposable`. We could rely on settings clear (to annoy @timbussmann :D)

Reply:
won't this be registered with the DI container anyway?

Reply:
> I intuitively went with lazy initialization. In theory, we could also eager initialize it. It felt more natural but I'm happy to be swayed over.

what would the benefit of the lazy initialization be? In case this provider is used, we are basically guaranteed that the client will be accessed, right? So Using a lazy will only delay the initialization rather than avoiding unnecessary allocations in case it won't be needed?

Reply:
Yes you are right. Forgot that

Reply:
> what would the benefit of the lazy initialization be? In case this provider is used, we are basically guaranteed that the client will be accessed, right? So Using a lazy will only delay the initialization rather than avoiding unnecessary allocations in case it won't be needed?

I was more thinking about deferring the credential chain resolution to the latest point possible but maybe I'm overthinking things.